docs/modules/ROOT/pages/node_pattern.adoc

```diff
@@ -207,6 +262,18 @@ Imagine you want to check if the number is `odd?` and also positive numbers:
 
 `(int [odd? positive?])` - is an int and the value should be odd and positive.
 
+NOTE: Refer to <<Predicate methods>> to see how `odd?` works.
+
+== `!` for Negation
+
+Node pattern `(send nil? :sum !int _)` would match a `sum` call where the first argument is *not* a literal integer.
+E.g.:
+
+* it will match `sum(2.0, 3)`, as the first argument is of a `float` type
+* it will not match `sum(2, 3)`, as the first argument is of an `int` type
+
+NOTE: Negation operator works with other node pattern syntax elements, `{}`, `[]`, `()`, `$`, but not with `<>`.
```

Comment:
I'm uncertain if this is intentional.
`(send nil? :foo <int ...>)` would work to match a `foo` call with at least one integer literal.
`(send nil? :foo !<int ...>)` would not work to match a `foo` call with no literal integer arguments. Is there a more canonical way to achieve that?

Reply:
Actually it's more complex than that. You can negate a pattern that targets a single element (e.g. `!{int | sym}`) but not other pattern (e.g. `!{int int | sym sym}`).
Not sure why you wrote it doesn't work with `$`

Reply:
Got it.
I understand that with variable number of elements like `{int | sym sym}` and `<int ...>` it's an easily justifiable behaviour.

I guess !<true false>` is more tricky due to the hidden number of elements ambiguity? It seems like it would match if there are no arguments (`foo()`), a single non-boolean argument (`foo(1)`), and multiple arguments as well, which just would open space for more confusion.

> Not sure why you wrote it doesn't work with `$`

Because I didn't! 😄

Thanks for the clarification! I've updated this section.